### PR TITLE
[Azure.Communication.CallAutomation] Add Channel Affinity to Recording API

### PR DESCRIPTION
**Pull request by williamzhao87**
Add Channel Affinity to StartRecording API to support Unmixed Audio
Also added autogenerated models for recognize

# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/communication/Azure.Communication.CallingServer/api/Azure.Communication.CallingServer.netstandard2.0.cs [hunk 1]
```diff
@@ -238,6 +237,19 @@ public partial class CallTransferFailed : Azure.Communication.CallingServer.Call
         public Azure.Communication.CallingServer.ResultInformation ResultInfo { get { throw null; } }
         public static Azure.Communication.CallingServer.CallTransferFailed Deserialize(string content) { throw null; }
     }
+    public partial class ChannelAffinity
+    {
+        public ChannelAffinity() { }
+        public int? Channel { get { throw null; } set { } }
```
Comment (cochi2) [inline sdk/communication/Azure.Communication.CallingServer/api/Azure.Communication.CallingServer.netstandard2.0.cs:243]: does this mean that Channel is optional?

File: sdk/communication/Azure.Communication.CallingServer/src/CallRecording.cs [hunk 2]
```diff
@@ -47,26 +49,32 @@ protected CallRecording()
...
         /// <param name="cancellationToken">The cancellation token.</param>
-        public virtual Response<RecordingStatusResult> StartRecording(CallLocator callLocator, Uri recordingStateCallbackEndpoint, StartRecordingOptions options = default, CancellationToken cancellationToken = default)
+        public virtual Response<RecordingStatusResult> StartRecording(StartRecordingOptions options, CancellationToken cancellationToken = default)
         {
             using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(CallRecording)}.{nameof(StartRecording)}");
             scope.Start();
             try
             {
-                StartCallRecordingRequestInternal request = new StartCallRecordingRequestInternal(CallLocatorSerializer.Serialize(callLocator))
+                StartCallRecordingRequestInternal request = new(CallLocatorSerializer.Serialize(options?.CallLocator))
```
Comment (cochi2) [inline sdk/communication/Azure.Communication.CallingServer/src/CallRecording.cs:60]: options cannot be optional, we should add a check for it.

File: sdk/communication/Azure.Communication.CallingServer/src/Models/ChannelAffinity.cs [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+namespace Azure.Communication.CallingServer
+{
+    /// <summary>
+    /// Channel affinity for a participant.
+    /// </summary>
+    public class ChannelAffinity
+    {
+        /// <summary> Channel number to which bitstream from a particular participant will be written. </summary>
+        public int? Channel { get; set; }
```
Comment (cochi2) [inline sdk/communication/Azure.Communication.CallingServer/src/Models/ChannelAffinity.cs:12]: does this mean that Channel is optional? what is the use of this, if channel is optional?

File: sdk/communication/Azure.Communication.CallingServer/src/Models/DtmfOptionsInternal.cs [hunk 4]
```diff
@@ -0,0 +1,18 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Core;
+
+namespace Azure.Communication.CallingServer
+{
+    [CodeGenModel("DtmfOptions")]
+
+    internal partial class DtmfOptionsInternal
```
Comment (cochi2) [inline sdk/communication/Azure.Communication.CallingServer/src/Models/DtmfOptionsInternal.cs:10]: why is this part of this PR?

File: sdk/communication/Azure.Communication.CallingServer/src/Models/StartRecordingOptions.cs [hunk 5]
```diff
@@ -1,13 +1,35 @@
...
-    /// Optional parameters for the Start Recording operation.
+    /// Options for the Start Recording operation.
     /// </summary>
     public class StartRecordingOptions
     {
+        /// <summary>
+        /// Parameters for the Start Recording operation.
+        /// </summary>
+        /// <param name="callLocator"> . </param>
+        public StartRecordingOptions(CallLocator callLocator)
```
Comment (cochi2) [inline sdk/communication/Azure.Communication.CallingServer/src/Models/StartRecordingOptions.cs:18]: CallBack URI has to be a required parameter as well.

File: sdk/communication/Azure.Communication.CallingServer/src/Models/StartRecordingOptions.cs [hunk 6]
```diff
@@ -1,13 +1,35 @@
...
+
+        /// <summary>
+        /// The callLocator.
+        /// </summary>
+        internal CallLocator CallLocator { get; }
+
+        /// <summary>
+        /// The callLocator.
+        /// </summary>
+        public Uri RecordingStateCallbackEndpoint { get; set; }
```
Comment (cochi2) [inline sdk/communication/Azure.Communication.CallingServer/src/Models/StartRecordingOptions.cs:31]: remove the setter from this.

File: sdk/communication/Azure.Communication.CallingServer/src/Models/DtmfOptionsInternal.cs (hunk 4 above)
Comment (williamzhao87) [reply sdk/communication/Azure.Communication.CallingServer/src/Models/DtmfOptionsInternal.cs:10]: The code was already merged to master, thus it got autogenerated into the swagger and these models were generated by Autorest.

File: sdk/communication/Azure.Communication.CallingServer/src/Models/ChannelAffinity.cs (hunk 3 above)
Comment (williamzhao87) [reply sdk/communication/Azure.Communication.CallingServer/src/Models/ChannelAffinity.cs:12]: Good catch, will make it non-nullable

File: sdk/communication/Azure.Communication.CallingServer/src/Models/StartRecordingOptions.cs (hunk 5 above)
Comment (williamzhao87) [reply sdk/communication/Azure.Communication.CallingServer/src/Models/StartRecordingOptions.cs:18]: As discussed, the callback Uri doesn't seem to be required on the PMA side

File: sdk/communication/Azure.Communication.CallingServer/src/Models/StartRecordingOptions.cs (hunk 6 above)
Comment (williamzhao87) [reply sdk/communication/Azure.Communication.CallingServer/src/Models/StartRecordingOptions.cs:31]: As discussed, the callback Uri doesn't seem to be required on the PMA side

File: sdk/communication/Azure.Communication.CallingServer/src/CallRecording.cs (hunk 2 above)
Comment (williamzhao87) [reply sdk/communication/Azure.Communication.CallingServer/src/CallRecording.cs:60]: Added a check

**Review by minwoolee-msft — APPROVED** [review]

**Pull request merged**